unittests.py

```diff
@@ -0,0 +1,199 @@
+"""Unit tests to check the maths functions."""
+
+import logging
+import numpy as np
+import unittest
```

Comment:
for the tests, could you do one test file per python module? It would probably create less dependencies. Here in order to run this test you end up needing matplotlib since you import visualizations